### PR TITLE
jenv: Adding caveats with general FAQ

### PR DESCRIPTION
**Pull request by pr4bh4sh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
audit doesn't pass, error below

```
> $ brew audit --strict jenv
jenv:
  * Use `depends_on :java` to set JAVA_HOME
Error: 1 problem in 1 formula detected

> $ echo $JAVA_HOME 
/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home
```

### CODE REVIEW

**Comment by SMillerDev** [comment]
This looks like something that should be in the manual for jenv and isn't specific to homebrew. Is that correct?

**Comment by pr4bh4sh** [comment]
@SMillerDev Yes, I guess apart from 
```
To activate jenv, add the following to your .zshrc:
         export PATH="$HOME/.jenv/bin:$PATH"
        eval "$(jenv init -)"
```

everything else could be removed.

**Comment by fxcoudert** [comment]
Even that part does not appear to be specific to Homebrew. It should go in the software's general documentation, not as a `caveat`.

**Comment by claui** [comment]
> Even that part does not appear to be specific to Homebrew. It should go in the software's general documentation, not as a `caveat`.

@fxcoudert I agree with you technically, though there are several dozen similar cases in homebrew-core ([examples](https://github.com/Homebrew/homebrew-core/search?q="your+bashrc"&type=Code)), which I’ve always appreciated whenever I had my user hat on. I’m in favour of the caveat because it’s user friendly.

Also, `*rc` settings tend to be stable over time; I don’t exactly feel they are a maintenance burden.

**Review by claui — CHANGES_REQUESTED** [review]

File: Formula/jenv.rb [hunk 1]
```diff
@@ -13,6 +13,15 @@ def install
     bin.write_exec_script libexec/"bin/jenv"
   end
 
+  def caveats
+    <<~EOS
+      To activate jenv, add the following to your .zshrc:
```
[the remark this answers is not in the document]
Comment (pr4bh4sh) [reply Formula/jenv.rb]: fixed

**Review by claui — COMMENTED** [review]
Thanks for the update @pr4bh4sh.

LGTM as is but deferring approval in order to collect more feedback from maintainers.

**Review by claui — APPROVED** [review]

**Comment by pr4bh4sh** [comment]
@claui Thanks for reviewing it. 
I was helping my colleague set up his machine and fought for good 1/2 an hour in figuring out why `jenv` is not working.

IMO End users(myself included) are usually as bright as sunshine on Pluto.

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**